ports/stm32/machine_i2s.c

```diff
@@ -566,6 +565,21 @@ STATIC bool i2s_init(machine_i2s_obj_t *self) {
         const pin_af_obj_t *af = pin_find_af(self->sd, AF_FN_I2S, self->i2s_id);
         GPIO_InitStructure.Alternate = (uint8_t)af->idx;
         HAL_GPIO_Init(self->sd->gpio, &GPIO_InitStructure);
+
+        #if defined(STM32H7)
+        // Make sure IOSWP bit is set correctly based on pin and mode
+        const pin_af_obj_t *pin_af = pin_find_af(self->sd, AF_FN_I2S, self->i2s_id);
+        if (pin_af != NULL) {
+            if ((pin_af->type == AF_PIN_TYPE_I2S_SDI
+                    && (self->mode == I2S_MODE_MASTER_RX || self->mode == I2S_MODE_SLAVE_RX))
```

Comment:
What is the purpose of the slave mode macros, I2S_MODE_SLAVE_RX, I2S_MODE_SLAVE_TX?

Reply:
I have another patch for preliminary I2S slave support on stm32, and this is needed for that to work.  I should put the slave PR up before this one...